src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -428,6 +428,17 @@ public void Initialize(ITelemetry telemetry)
                 var version = LazyInitializer.EnsureInitialized(ref this.sdkVersion, this.GetSdkVersion);
                 telemetry.Context.Internal.SdkVersion = version;
             }
+
+            // set RoleInstance and NodeName to the machine name if they're not initialized yet
+            if (string.IsNullOrEmpty(telemetry.Context.Internal.NodeName))
+            {
+                telemetry.Context.Internal.NodeName = PlatformSingleton.Current.GetMachineName();
```

Comment:
1) added try/catch in PlatformImplementation
2) yes, it's intentional - i don't want to rely on customers choice for role instance as it may be done for display purposes and conflict with billing